src/pages/pupil/single-course/PupilCourseButtons.tsx

```diff
@@ -2,19 +2,33 @@ import { ApolloQueryResult } from '@apollo/client';
 import { Button, Modal, Stack, useTheme, useToast, VStack } from 'native-base';
 import { useCallback, useEffect, useMemo, useState } from 'react';
 import { useTranslation, Trans } from 'react-i18next';
-import { Subcourse } from '../../../gql/graphql';
+import { Chat_Type } from '../../../gql/graphql';
 import { useLayoutHelper } from '../../../hooks/useLayoutHelper';
 import CourseConfirmationModal from '../../../modals/CourseConfirmationModal';
 import { getTrafficStatus } from '../../../Utility';
 import WaitinglistBanner from '../../../widgets/WaitinglistBanner';
 import JoinMeeting from '../../subcourse/JoinMeeting';
 import AlertMessage from '../../../widgets/AlertMessage';
+import OpenSubcourseChat from '../../subcourse/OpenSubcourseChat';
 
 type CanJoin = {
     allowed: boolean;
     reason?: 'not-participant' | 'no-lectures' | 'already-started' | 'already-participant' | 'grade-to-low' | 'grade-to-high' | 'subcourse-full' | null;
 };
 
+type SubcourseOfPupil = {
+    id: number;
+    participantsCount: number;
+    maxParticipants: number;
+    isParticipant: boolean;
+    isOnWaitingList: boolean;
+    canContactInstructor: { allowed: boolean };
+    conversationId?: string | null | undefined;
+    allowChatContactProspects?: boolean;
+    allowChatContactParticipants?: boolean;
+    groupChatType: Chat_Type;
+};
```

Comment:
What about something like:
```suggestion
type SubcourseOfPupil = Subcourse & {
    groupChatType: Chat_Type;
};
```
This approach helps you maintain the types better, if `Subcourse` changes, you're just using the new type instead of having to extend it.

Reply:
Why was this resolved?